docs/modules/ROOT/pages/adr/0030-function-revisions.adoc

```diff
@@ -0,0 +1,176 @@
+= ADR 0030 - Function Revisions
+:adr_author:    Simon Beck
+:adr_owner:     Schedar
+:adr_reviewers:
+:adr_date:      2025-01-29
+:adr_upd_date:  2025-01-29
+:adr_status:    draft
+:adr_tags:      comp-functions,crossplane
+
+include::partial$adr-meta.adoc[]
+
+[NOTE]
+.Summary
+====
+In order to make rollouts easier, we want to make sure any changes will only be applied to existing instances during their respective maintenance window.
+====
+
+PoC URL: https://github.com/vshn/component-appcat/pull/612
+
+== Context
+Currently, if we have changes that trigger restarts, then we have either to:
+
+* time them during the cluster maintenance
+* disable ArgoCD and manually apply Maintenance hooks to re-enable it again
+
+Both of these solutions are manual and prone to error. Additionally, the first approach breaks our Renovate process that deploys a change on test clusters for a week before its rolled out everywhere.
+
+Crossplane has a concept called https://docs.crossplane.io/latest/concepts/composition-revisions/#using-composition-revisions[Composition Revisions]. They are automatically created each time a change in a composition is applied. By default, any claim or composite deployed will automatically point to the most recent revision as soon as they are available.
```

Comment:
Are we using composition revisions already today? I think we should point out this fact in this description. Because if we aren't already using them, I'd point out very prominently that this design introduces the usage of them. Right now, I only ready "between the lines" that we're introducing that concept because of function revisions.

Reply:
We're not using them as of now. I'll add something to state that more explicit.